chia/rpc/wallet_rpc_api.py

```diff
@@ -1370,6 +1370,9 @@ async def cat_spend(self, request, hold_lock=True) -> EndpointResult:
             puzzle_hashes.append(decode_puzzle_hash(request["inner_address"]))
             if "memos" in request:
                 memos.append([mem.encode("utf-8") for mem in request["memos"]])
+        coins = None
+        if "coins" in request and len(request["coins"]) > 0:
+            coins = set([Coin.from_json_dict(coin_json) for coin_json in request["coins"]])
```

Comment:
I take it this is one of the CLI bugs this PR is supposed to fix. is that right?
is `cat_spend()` an internal function of the `wallet_rpc_api.py`? If so, it makes me uneasy to pass the arguments as a `dict`. If not (and it's public), the types should be validated. e.g. `request["coins"]` could be a string, but you'd end up iterating over characters instead of json entries.

The fact that `request` is allowed to be untyped suggests that this whole file doesn't have mypy turned up to max yet either, which is another cause for concern (genrally)

Reply:
cat_spend is not an internal function, it is a usable rpc.